src/LanguageServer.php

```diff
@@ -167,11 +167,12 @@ public function __construct(ProtocolReader $reader, ProtocolWriter $writer)
      * @param ClientCapabilities $capabilities The capabilities provided by the client (editor)
      * @param string|null $rootPath The rootPath of the workspace. Is null if no folder is open.
      * @param int|null $processId The process Id of the parent process that started the server. Is null if the process has not been started by another process. If the parent process is not alive then the server should exit (see exit notification) its process.
+     * @param mixed $initializationOptions The options send from client to initialize the server
      * @return Promise <InitializeResult>
      */
-    public function initialize(ClientCapabilities $capabilities, string $rootPath = null, int $processId = null): Promise
+    public function initialize(ClientCapabilities $capabilities, string $rootPath = null, int $processId = null, $initializationOptions = null): Promise
```

Comment:
The actual options are taken from the configuration and are just passed with the initialize method. felixfbecker/vscode-php-intellisense@59e6f2cc2f7047bf3990d7463ab546c89eb330da

So you want to move the indexing to `onDidChangeConfiguration`? Otherwise you would index 2x, because this event is sent after the intialize (by vscode).